gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageFileSystem.java

```diff
@@ -629,11 +630,15 @@ public void rename(URI src, URI dst) throws IOException {
       StorageResourceId dstResourceId =
           StorageResourceId.fromUriPath(dst, /* allowEmptyObjectName= */ true);
 
-      gcs.copy(
-          srcResourceId.getBucketName(), ImmutableList.of(srcResourceId.getObjectName()),
-          dstResourceId.getBucketName(), ImmutableList.of(dstResourceId.getObjectName()));
+      StorageResourceId destinationObject =
+          new StorageResourceId(
+              dstResourceId.getBucketName(),
+              dstResourceId.getObjectName(),
+              /* generationId= */ 0); // set generationId = 0 to ensure no live version on gcs
```

Comment:
We create 2 `StorageResourceId`s (`dstResourceId` and `destinationObject`) for destination object here, can we merge it into one?